### PR TITLE
refactor: Create a separated structure for build-system

### PR DESCRIPTION
**Pull request by xmnlab**
## How to test these changes

* ```...```

## Pull Request checklists

This PR is a:
- [ ] bug-fix
- [ ] new feature
- [ ] maintenance

About this PR:
- [ ] it includes tests.
- [ ] the tests are executed on CI.
- [ ] the tests generate log file(s) (path).
- [ ] pre-commit hooks were executed locally.
- [ ] this PR requires a project documentation update.

Author's checklist:
- [ ] I have reviewed the changes and it contains no misspelling.
- [ ] The code is well commented, especially in the parts that contain more complexity.
- [ ] New and old tests passed locally.

## Reviewer's checklist

Copy and paste this template for your review's note:

```
## Reviewer's Checklist

- [ ] I managed to reproduce the problem locally from the `main` branch
- [ ] I managed to test the new changes locally
- [ ] I confirm that the issues mentioned were fixed/resolved .
```

### CODE REVIEW

**Comment by xmnlab** [comment]
@Anavelyz @YurelyCamacho for now I think that would be the approach.
I still need to finish that, but I need to leave for now.
sorry for the delay, I tried a bunch of approaches .. but none worked here :(

In order to move this forward faster, let's try this approach for now. and I will revisit it next week.

File: src/scicookie/{{cookiecutter.project_slug}}/build-system/poetry-pyproject.toml [hunk 1]
```diff
@@ -112,71 +82,5 @@ pandoc = "^2.3"
...
-]
 
-[tool.ruff.pydocstyle]
-convention = "numpy"
-{% endif %}
-{%- if cookiecutter.use_mypy == "yes" %}
-[tool.mypy]
-no_strict_optional = false
-{% endif %}
-{#- keep this line at the end of the file -#}
```
Comment (YurelyCamacho) [inline src/scicookie/{{cookiecutter.project_slug}}/build-system/poetry-pyproject.toml:182]: @xmnlab I think this line should go at the end of the file, there is no empty line at the end. That's what I can see

**Review by YurelyCamacho — COMMENTED** [review]
@xmnlab I left a comment to see if that might be what's going wrong. I understood what you wanted to do with the structure. Thanks

File: src/scicookie/{{cookiecutter.project_slug}}/build-system/poetry-pyproject.toml (hunk 1 above)
Comment (xmnlab) [reply src/scicookie/{{cookiecutter.project_slug}}/build-system/poetry-pyproject.toml:182]: thanks @YurelyCamacho , I will revisit that again now! appreciate your review!

**Comment by xmnlab** [comment]
Thanks @YurelyCamacho that did the trick :)

**Pull request merged**